inst/examples/RcppProgressETA/src/eta_progress_bar.hpp

```diff
@@ -19,141 +19,168 @@
 
 // for unices only
 #if !defined(WIN32) && !defined(__WIN32) && !defined(__WIN32__)
-#include <Rinterface.h>
+  #include <Rinterface.h>
 #endif
 
 class ETAProgressBar: public ProgressBar{
-public: // ====== LIFECYCLE =====
-
-	/**
-	 * Main constructor
-	 */
-	ETAProgressBar()  {
-		_max_ticks = 50;
-		_ticks_displayed = 0;
-		_finalized = false;
-		_mode_flag = 1;
-		_timer_flag = true;
-		_timer_count = 1;
-	}
-
-	~ETAProgressBar() {
-	}
-
-public: // ===== main methods =====
-
-	void display() {
-		REprintf("0%% 10%%\n");
-		REprintf("|----|\n");
-		flush_console();
-	}
-
-	// will finalize display if needed
-	void update(float progress) {
-
-    _update_ticks_display(progress);
-    if (_ticks_displayed >= _max_ticks)
-      _finalize_display();
-
-		if (_timer_flag) {
-		  _timer_flag = false;
-		  time(&start);
-		} else if (_ticks_displayed >= _max_ticks) {
-		  time(&end);
-		} else if ((int) (((double)_ticks_displayed / (double)_max_ticks) * 100) == 10 * _timer_count) {
-
-      _timer_count++;
-
-		  time(&end);
-
-		  double pas_time = std::difftime(end, start);
-		  double rem_time = (pas_time / progress) * (1 - progress);
-		  int pas_time_simple = (int) pas_time;
-      int rem_time_simple = (int) rem_time;
-
-      std::stringstream rem_strs;
-      if (rem_time_simple > 3600) {
-        rem_strs << (int) (rem_time_simple / 3600) << "h";
-      } else if (rem_time_simple > 60) {
-        rem_strs << (int) (rem_time_simple / 60) << "min";
-      } else {
-        rem_strs << rem_time_simple << "s";
+  public: // ====== LIFECYCLE =====
+    
+    /**
+    * Main constructor
+    */
+    ETAProgressBar()  {
+      _max_ticks = 50;
+      _ticks_displayed = 0;
+      _finalized = false;
+      _mode_flag = 1;
+      _timer_flag = true;
+    }
+    
+    ~ETAProgressBar() {
+    }
+    
+    public: // ===== main methods =====
+      
+      void display() {
+        REprintf("0%%   10   20   30   40   50   60   70   80   90   100%%\n");
+        REprintf("[----|----|----|----|----|----|----|----|----|----|\n");
+        flush_console();
       }
-      std::string rem = rem_strs.str();
-
-      std::stringstream pas_strs;
-      if (pas_time_simple > 3600) {
-        pas_strs << (int) (pas_time_simple / 3600) << "h";
-      } else if (pas_time_simple > 60) {
-        pas_strs << (int) (pas_time_simple / 60) << "min";
-      } else {
-        pas_strs << pas_time_simple << "s";
+      
+      // will finalize display if needed
+      void update(float progress) {
+        
+        // condition to manage time measurement
+        if (_timer_flag) {
+          _timer_flag = false;
+          time(&start);
+        } else if ((progress > 0.001) & !_finalized) {
```

Comment:
conditions are too complex to follow. The function should start by checking _finalized and returning immediatelyif it is true. You should also comment about why the  progress > 0.001 ?

Reply:
I added `if (_finalized) return;` before the condition, simplified the condition and removed the artificial delay. I can't reconstruct why I added it in the first place.